pkg/api/meta/restmapper.go

```diff
@@ -148,6 +148,23 @@ func kindToResource(kind string, mixedCase bool) (plural, singular string) {
 	return
 }
 
+// ResourceSingularlizer converts a resource name from plural to singular (e.g., from pods to pod)
+func ResourceSingularlizer(resource string) (singular string) {
+	if len(resource) == 0 {
+		return
+	}
+	if strings.HasSuffix(resource, "ies") {
```

Comment:
`strings.ToLower` just in case not all callers are perfect?  You'd probably want to switch the case back to match in the output.